scheduler/scheduler-server/src/test/java/functionaltests/job/TestExecuteScriptsOnBusyNode.java

```diff
@@ -0,0 +1,314 @@
+        }
+
+    }
+
+    /**
+     * Starts an scheduler in non-fork mode with an empty rm
+     *
+     * @throws Exception
+     */
+    @BeforeClass
+    public static void startDedicatedScheduler() throws Exception {
+        schedulerHelper = new SchedulerTHelper(false,
+                                               new File(SchedulerTHelper.class.getResource("/functionaltests/config/scheduler-nonforkedscripttasks.ini")
+                                                                              .toURI()).getAbsolutePath(),
+                                               null,
+                                               null);
+    }
+
+    @Before
+    public void deployNode() throws Exception {
+        startNode();
+        executorService = Executors.newSingleThreadExecutor();
+    }
+
+    @Test
+    public void testExecuteScriptOnBusyNode() throws Exception {
+        FileLock fileLock = new FileLock();
+        Path fileLockPath = fileLock.lock();
+
+        log("Submit job");
+        final JobId jobId = schedulerHelper.submitJob(createJobWithLock(fileLockPath.toString()));
+
+        log("Wait when node becomes busy");
+        RMNodeEvent event;
+        do {
+            event = schedulerHelper.waitForAnyNodeEvent(RMEventType.NODE_STATE_CHANGED, TIMEOUT);
+        } while (!event.getNodeState().equals(NodeState.BUSY));
+
+        log("Wait when task starts");
+        schedulerHelper.waitForEventTaskRunning(jobId, "Test task");
+
+        log("Wait some time until task launcher is started");
+        Thread.sleep(5000);
+
+        List<ScriptResult<Object>> results = schedulerHelper.getResourceManager()
+                                                            .executeScript("println \"Executing Script\";",
+                                                                           "groovy",
+                                                                           TargetType.NODE_URL.name(),
+                                                                           Sets.newSet(testNode.getNodeURL()));
+        Assert.assertEquals(1, results.size());
+        ScriptResult result = results.get(0);
+        if (result.errorOccured()) {
+            result.getException().printStackTrace();
+            Assert.fail("Script execution failed");
+        }
+
+        log("Let task finish");
+        fileLock.unlock();
+
+        log("Wait when job finish");
+        schedulerHelper.waitForEventJobFinished(jobId, TIMEOUT);
+
+        schedulerHelper.waitUntilState(testNode.getNodeURL(), NodeState.FREE, TIMEOUT);
+
+        log("Check job result");
+        checkJobResult(schedulerHelper.getSchedulerInterface(), jobId);
+        checkActiveObjects();
+    }
+
+    private void checkActiveObjects() throws NodeException, ActiveObjectCreationException, InterruptedException {
+        try {
+            // This is to avoid slight asynchronous delays in active objects termination
+            Thread.sleep(1000);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this use of "Thread.sleep()". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2925)